libs/ha/ha_common_libs_k8s.py

```diff
@@ -1795,9 +1805,22 @@ def iam_bucket_cruds(self, event, s3_obj=None, user_crud=False, num_users=None,
                 if len(del_users) > i_i:
                     LOGGER.debug("Deleting %s user", del_users[i_i])
                     user = del_users[i_i]
-                    resp = self.delete_s3_acc_buckets_objects({user: del_users_dict[user]})
-                    if not resp[0]:
-                        user_del_failed.append(user)
+                    try:
+                        if not header:
+                            resp = self.delete_s3_acc_buckets_objects({user: del_users_dict[user]})
+                        else:
+                            resp = self.delete_iam_user_with_header({user: del_users_dict[user]},
+                                                                    header)
+                        if not resp[0]:
+                            user_del_failed.append(user)
+                            if event.is_set():
+                                exp_fail.append(user)
+                            else:
+                                failed.append(user)
+                    except (req_exception.ConnectionError, req_exception.ConnectTimeout) as error:
+                        LOGGER.exception("Error: %s", error)
```

Comment:
This handling is not needed now as you are already handling exceptions in delete_iam_user_with_header.